### DISCUSSION TITLE
DenseGCNConv with to_hetero

### DISCUSSION THREAD

**Discussion by wgeul** [pyg-team/pytorch_geometric#4764] [Q&A] [answered]
Hi,

Is it possible to convert `DenseGCNConv` or other layers that have a forward function that only accepts `x: Tensor` by default to a heterogeneous implementation? Is there by any chance a quick solution to support `x_src` and `x_dest` for such pre-defined layers?

I'm trying to make a model that performs a Graph-level regression task on a heterogeneous graph network and think the Dense layers are most appropriate for the task.

**Comment by Padarn** [comment]
Does https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/to_hetero_transformer.py help solve your problem?

It allows you to build a heterogenous model out of homogenous modules. There is an example of it's use here https://github.com/pyg-team/pytorch_geometric/blob/8bd9ae484369192af62a040540d53d6bfc881001/examples/hetero/to_hetero_mag.py#L49

**Comment by rusty1s** [comment, marked as answer]
It is currently not possible, I am sorry. Even for graph-level tasks, we recommend to use sparse representations.

**Reply by Padarn** [reply]
+1, sorry I originally misunderstood the question

**Reply by wgeul** [reply]
Thanks for the answers.
